src/modules/todos/controllers.js

```diff
@@ -1,144 +1,154 @@
-			statcode: 201,
-			data: {
-				id: todo.id,
-				userId: todo.userId,
-				title: todo.title,
-				status: todo.status,
-				dueDate: todo.dueDate,
-				createdAt: todo.createdAt,
-				updatedAt: todo.updatedAt,
-			}
-		})
+			statuscode: 201,
+			data: newTodo.rows[0],
+		});
 	} catch (err) {
 		next(err);
-	};
-};
-
-export function getTodo(req, res) {
-	const id = req.params.id;
-	const findTodo = todos.find((todo) => todo.id === id);
-	if (!findTodo) {
-		throw new AppException(404, "Unable to retrieve todo")
 	}
-	else if (findTodo.userId === req.userId) {
-		return res.status(200).json({
-			statusCode: 200,
-			data: {
-				id: findTodo.id,
-				userId: findTodo.userId,
-				title: findTodo.title,
-				status: findTodo.status,
-				dueDate: findTodo.dueDate,
-				createdAt: findTodo.createdAt,
-				updatedAt: findTodo.updatedAt,
-			},
-		});
-	}
-	else {
-		throw new AppException(403, "Unauthorized")
+}
+
+export async function getTodo(req, res, next) {
+	try {
+		const { id } = req.params;
+		const findTodo = await pool.query(queries.findTodo, [id]);
+		if (!findTodo.rowCount) {
+			throw new AppException(404, "Unable to retrieve todo");
+		} else if (findTodo.rows[0].user_id === req.userId) {
+			return res.status(200).json({
+				statusCode: 200,
+				data: findTodo.rows[0],
+			});
+		} else {
+			throw new AppException(403, "Unauthorized");
+		}
+	} catch (err) {
+		next(err);
 	}
-};
+}
 
-export function updateTodo(req, res) {
-	let today = new Date();
-		
-	let strDate = 'Y-m-d h:M:s'
-		.replace('Y', today.getFullYear())
-		.replace('m', today.getMonth()+1)
-		.replace('d', today.getDate())
-		.replace('h', today.getHours())
-		.replace('M', today.getMinutes())
-		.replace('s', today.getSeconds());
+export async function updateTodo(req, res, next) {
+	try {
+		let today = new Date();
 
-	const id = req.params.id;
-	const title = req.body.title;
-	const status = req.body.status
-	const dueDate = req.body.dueDate;
-	const updateTodo = todos.find((todo) => todo.id === id);
-	if (!updateTodo) {
-		throw new AppException(404, "Unable to retrieve todo")
-	}
-	else if (updateTodo.userId === req.userId) {
-		if (title) {
-			updateTodo.title = title;
-		}
-		if (status) {
-			updateTodo.status = status;
-		}
-		if (dueDate) {
-			updateTodo.dueDate = dueDate;
+		let strDate = "Y-m-d h:M:s"
+			.replace("Y", today.getFullYear())
+			.replace("m", today.getMonth() + 1)
+			.replace("d", today.getDate())
+			.replace("h", today.getHours())
+			.replace("M", today.getMinutes())
+			.replace("s", today.getSeconds());
+
+		const { id } = req.params;
+		const { title, status, dueDate } = req.body;
+		const updatedAt = strDate;
+		const findTodo = await pool.query(queries.findTodo, [id]);
+		if (!findTodo.rowCount) {
+			throw new AppException(404, "Unable to retrieve todo");
+		} else if (findTodo.rows[0].user_id === req.userId) {
+			if (title) {
+				const updateTitle = await pool.query(queries.updateTitle, [
```

Comment:
Updating things one at a time like this is not a good idea. You can get a bit more clever with it. Using string concatenation, you can build the query based on the data the user passed. Here is an example of what it would look like.

```javascript
let updateQuery = "UPDATE todos SET ";

["title", "status", "dueDate"].foreach(key => {
    if (key in req.body) {
        updateQuery += `${key} = ?`
    }
});

updateQuery += " WHERE id = ?";
```

This is just a rough example. I did not test the code.

Reply:
@Abiodun-Shittu Here is what it will look like it I were trying to update a todo.

```javascript
const todoId = 1;
const body = {
    title: "New Item",
    status: "done",
    dueDate: "2022-10-20",
};

const updateKeys = Object.keys(body);
let updateQuery = "UPDATE todos SET";
const valuesInOrder = [];

updateKeys.forEach((key, index) => {
    const value = body[key];
    valuesInOrder.push(value);

    updateQuery += ` ${key} = ?`;
    if (index < updateKeys.length - 1) {
        updateQuery += ", "
    }
});
valuesInOrder.push(todoId);

updateQuery += " WHERE id = ?";

// console.log(updateQuery);
pool.query(updateKey, valuesInOrder);
```